src/key-container/local-storage-container.js

```diff
@@ -44,55 +46,235 @@ class LocalStorageContainer {
+      this.db.insert(`${this.prefix}keySeed`, JSON.stringify({ keySeedEncrypted, pathKeySeedEncrypted }));
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Gets key chain and its current path
+   * @return {Object} - Contains key chain and its current path
+   */
+  getKeySeed() {
+    if (this.isUnlock()) {
+      const keySeedDb = this.db.listKeys(`${this.prefix}keySeed`);
+      if (keySeedDb.length === 0) {
+        return undefined;
+      }
+      const { keySeedEncrypted, pathKeySeedEncrypted } = JSON.parse(this.db.get(keySeedDb));
+      const keySeed = kcUtils.decrypt(this.encryptionKey, keySeedEncrypted);
+      const pathKeySeed = kcUtils.decrypt(this.encryptionKey, pathKeySeedEncrypted);
+      const pathKey = (utils.hexToBytes(pathKeySeed)).readUInt32BE();
+      return { keySeed, pathKey };
+    }
+    return undefined;
+  }
+
+  /**
+   * Increase one level on key chain path
+   * @return {Bool} - Acknowledge
+   */
+  increaseKeyPath() {
+    if (this.isUnlock()) {
+      const { keySeed, pathKey } = this.getKeySeed();
+      const increasePathKey = pathKey + 1;
+      const pathKeyDb = Buffer.alloc(4);
+      pathKeyDb.writeUInt32BE(increasePathKey);
+      const keySeedEncrypted = kcUtils.encrypt(this.encryptionKey, keySeed);
+      const pathKeySeedEncrypted = kcUtils.encrypt(this.encryptionKey, utils.bytesToHex(pathKeyDb));
+      this.db.insert(`${this.prefix}keySeed`, JSON.stringify({ keySeedEncrypted, pathKeySeedEncrypted }));
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Generates recovery seed and store it into the database
+   * @param {String} masterSeed - Master seed representes as a memonic
+   * @return {String} - Public recovery address
+   */
+  generateRecoveryAddr(masterSeed) {
+    if (this.isUnlock()) {
+      const root = hdkey.fromMasterSeed(masterSeed);
+      const pathRecovery = "m/44'/60'/1'";
+      const addrNodeRecovery = root.derive(pathRecovery);
+      const privRecovery = addrNodeRecovery._privateKey;
+      const addressRecovery = ethUtil.privateToAddress(privRecovery);
+      const addressRecoveryHex = utils.bytesToHex(addressRecovery);
+      const privRecoveryHex = utils.bytesToHex(privRecovery);
+      const privRecoveryHexEncrypted = kcUtils.encrypt(this.encryptionKey, privRecoveryHex);
+      this.db.insert(this.prefix + CONSTANTS.IDRECOVERYPREFIX + addressRecoveryHex, privRecoveryHexEncrypted);
+      return addressRecoveryHex;
+    }
+    return undefined;
+  }
+
+  /**
+   * Generates a key from a given key chain and its path
+   * Key is stored as: { 'public address': 'encrypted(private key)' }
+   * @param {Number} keyChainPath - First path to derive the key
+   * @param {Number} keyPath - Second path to derive the key
+   * @returns {String} Public address key generated
+   */
+  generateSingleKey(keyProfilePath, keyPath) {
+    if (this.isUnlock()) {
+      let path = "m/44'/60'/0'/";
+      path = `${path + keyProfilePath}/${keyPath}`;
```

Comment:
Why not let path  = `m/44'/60'/0'/${keyProfilePath}/${keyPath}`

Reply:
totally true :)